source/server/config_validation/server.cc

```diff
@@ -53,7 +54,9 @@ ValidationInstance::ValidationInstance(
                           store),
       mutex_tracer_(nullptr), grpc_context_(stats_store_.symbolTable()),
       http_context_(stats_store_.symbolTable()), router_context_(stats_store_.symbolTable()),
-      time_system_(time_system), server_contexts_(*this) {
+      time_system_(time_system), server_contexts_(*this),
+      drain_manager_(std::make_unique<Server::DrainManagerImpl>(
```

Comment:
What methods are ever called on this top-level drain manager owned by the server?  It seems to me that drains are initiated at the listener or filter chain levels.  Is this layer of managers needed or could the parent DrainManagerImpl be created directly by source/server/listener_impl.cc prior to this PR?

Reply:
There is a top-level drain-manager on the InstanceImpl that will initiate a server-wide drain-sequence via `drainListeners()`. Prior to this PR, this was the main drain-manager object. The validation server doesn't need a drain-manager since draining isn't necessarily something that it does.

This change is so that when it is constructing listeners and filter-chains, the child-drain manager creation bits don't fail where previously they were attempting to call the `drainManager()` function and hitting the `NOT_IMPLEMENTED_GCOVR_EXCL_LINE`.